filters/SMRFilter.cpp

```diff
@@ -119,6 +120,8 @@ void SMRFilter::addArgs(ProgramArgs& args)
     args.add("ignore", "Ignore values", m_args->m_ignored);
     args.add("returns", "Include last returns?", m_args->m_returns,
              {"last", "only"});
+    args.add("synthetic", "Allow synthetic returns?", m_args->m_allowSynthetic,
+             true);
```

Comment:
What about `withheld`? The user can't specify ignoring those currently given how our dimRange stuff works. Maybe put this in as a future TODO.

Reply:
This is something important for NZ national QC I'm trying to calculate if extreme scan angles are found in withheld flagged points vs non-withheld. If they are in non-withheld it could potentially impact our products e.g. CL2 point cloud files, DSM and DEM. Thank you again for an epic piece of open source software.

Reply:
I'm not sure what's up with this comment. The current code supports all the classification filter bits.

Reply:
pipelines I have tried using pdal 2.3.0: to filter out (1) unclassified points flagged as withheld;

1.
```
{
    "pipeline" : [
    {
        "type" : "readers.las",
        "filename" : "input.las"
    },
	{
		"type":"filters.range",
		"limits":"Classification![129:129]"
	},	
    {
        "type" : "writers.las",
        "a_srs" : "EPSG:2193+7839",
        "filesource_id" : "0",
        "filename" : "output.las"
    }
    ]
  }

```
2.
```
{
    "pipeline" : [
    {
        "type" : "readers.las",
        "filename" : "input.las"
    },
	{
		"type":"filters.range",
		"limits":"Classification![129:129]",
		"where":"Classification!=129"
	},	
    {
        "type" : "writers.las",
        "a_srs" : "EPSG:2193+7839",
        "filesource_id" : "0",
        "filename" : "output.las"
    }
    ]
  }
```
  
3. 
` pdal info --stats C:\Temp\scan_angle\test_file\CL2_BW24_2020_1000_4526.las --filters.range.limits=Classification![129:129] --filters.stats.dimensions=Classification --filters.stats.count=Classification`

 All my outputs have these points still intact in their files and the pdal info doesn't appear to support this filter.

Reply:
Filtering out withheld points is not related to this PR unless you're attempting to do this when using the SMRF filter.  If you're having a general issue with withheld points, you should post to the mailing list or open another PR if appropriate.

Reply:
Thank you for the heads up - I will post in pdal mailing list this query.